workflows/iuf/operations/management-nodes-rollout/management-storage-nodes-reboot.yaml

```
@@ -0,0 +1,361 @@
+            templateRef:
+              name: ncn-reboot-common
+              template: reboot-ncn
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+                - name: global_params
+                  value: "{{inputs.parameters.global_params}}"
+                - name: targetNcn
+                  value: "{{inputs.parameters.targetNcn}}"
+          - name: wait-for-cloud-init
+            dependencies:
+              - reboot-node
+            templateRef:
+              name: ncn-reboot-common
+              template: wait-for-cloud-init
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+                - name: targetNcn
+                  value: "{{inputs.parameters.targetNcn}}"
+          - name: wait-for-node-health
+            templateRef:
+              name: check-ceph-health
+              template: main
+            dependencies:
+              # check health once node reboot is complete
+              - wait-for-cloud-init
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: 'false'
+          - name: wait-for-cfs
+            dependencies: 
+              - wait-for-cloud-init
+            templateRef:
+              name: ncn-reboot-common
+              template: wait-for-cfs
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+                - name: targetNcn
+                  value: "{{inputs.parameters.targetNcn}}"
+          - name: platform-health-checks
+            dependencies: 
+              - wait-for-cfs
+              - wait-for-node-health
+            templateRef:
+              name: ncn-reboot-common
+              template: platform-health-checks
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+    - name: process-sets
+      inputs:
+        parameters:
+          - name: sets
+      script:
+        image: artifactory.algol60.net/csm-docker/stable/docker.io/alpine/git:2.32.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          sets="{{inputs.parameters.sets}}"
+          echo $sets > /tmp/sets
+          num_sets=$(echo $sets | wc -w)
+          echo $num_sets > /tmp/num_sets
+      outputs:
+        parameters:
+          - name: sets
+            valueFrom:
+              path: "/tmp/sets"
+          - name: num_sets
+            valueFrom:
+              path: "/tmp/num_sets"
+    - name: move-to-next-set
+      inputs:
+        parameters:
+          - name: count
+      script:
+        image: artifactory.algol60.net/csm-docker/stable/docker.io/alpine/git:2.32.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          current_count="{{inputs.parameters.count}}"
+          counter=$(($current_count + 1 ))
```

You don't need `$` to reference variables inside of `$((...))`
```suggestion
          counter=$((current_count + 1 ))
```